v5/patch.go

```diff
@@ -831,7 +851,18 @@ func (p Patch) remove(doc *container, op Operation, options *ApplyOptions) error
 		return errors.Wrapf(ErrMissing, "remove operation does not apply: doc is missing path: \"%s\"", path)
 	}
 
-	err = con.remove(key, options)
+	if key == "-" && options.SupportDeleteByValue {
+		value := op.value()
+		switch v := con.(type) {
+		case *partialArray:
+			err = v.removeByValue(value, options)
+		default:
+			return errors.Wrapf(ErrMissing, "remove operation does not apply: not an array: \"%s\"", path)
```

Comment:
I am not sure if `ErrMissing` is the right error here. Because the path exists but it is not an array.

Reply:
I kind of agree. But I was looking at the few errors that are used by this library, and this one seemed the most appropriate. It also seems to be used in some cases where it's not clear that "something is missing".